app/src/main/java/com/github/dedis/student20_pop/MainActivity.java

```diff
@@ -1,17 +1,58 @@
 package com.github.dedis.student20_pop;
 
-import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
 
 import android.os.Bundle;
+import android.view.View;
 
-/*
-    Activity used to display "Hello World"
- */
-public class MainActivity extends AppCompatActivity {
+import com.github.dedis.student20_pop.ui.ConnectFragment;
+import com.github.dedis.student20_pop.ui.HomeFragment;
+import com.github.dedis.student20_pop.ui.LaunchFragment;
+
+/**
+ * Activity used to display the different UIs
+**/
+public final class MainActivity extends FragmentActivity {
+
+    private static final String TAG = MainActivity.class.getSimpleName();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
```

Comment:
Where is this variable R defined? Seems to be appear in a lot of places

Reply:
https://stackoverflow.com/questions/4953077/what-is-the-class-r-in-android

Maybe this thread can help you :)

Reply:
R is an auto-generated class containing the resource IDs for all the resources of res/directory

Reply:
Ah I see, thank you @AlexisTabin and @celinecamacho! Can you please put that thread link in the project Readme so that someone else looking at the code and is not familiar with Android development can determine what this “R” variable is about. Thanks!